specs/snapshots.md

```diff
@@ -0,0 +1,91 @@
+    - `latest_params` is the last block height and hash the manager was notified about. If the manager is busy exporting the previous snapshot, it might skip some of these notifications, as it only reacts to them when it’s not already doing work.
+    - `current_download` is the state of any snapshot currently being fed to the manager by CometBFT, which is being stored in a temporary directory until all chunks are available.
+- [`SnapshotClient`](https://github.com/consensus-shipyard/ipc/blob/specs/fendermint/vm/snapshot/src/client.rs) shares its `SnapshotState` with the `SnapshotManager` and is the interface for the `App` to send notification and run queries over the state, as well as to save chunks coming from CometBFT to files and kick off the state restoration when all are present.
+- [`car`](https://github.com/consensus-shipyard/ipc/tree/specs/fendermint/vm/snapshot/src/car) module has utilities such as `ChunkWriter`, `BlockStreamer` and `split` to divide CAR files into chunks in a way that no IPLD `Block` is split across files.
+
+The following methods are worth looking at:
+
+- `SnapshotManager::run` to see how notifications are handled
+- `SnapshotManager::create_snapshot` to see steps involved in creating a snapshot:
+    - create a snapshot from a given block height and FVM parameters
+    - export the snapshot CAR file to a temporary directory
+    - record its size and create a checksum (these go into the manifest)
+    - split the CAR file into chunks with the configured target size
+    - create the manifest and export it to JSON
+    - move the temporary directory to the final place under the manager
+- `SnapshotClient::save_chunk` handles chunks from remote peers:
+    - checks that it’s the next expected chunk (CometBFT loads them sequentially)
+    - saves them to a part file
+    - if it’s the last chunk (according to the manifest), it runs the checksum and returns the `SnapshotItem` ready to be loaded
+- `App::apply_snapshot_chunk` calls `SnapshotClient::save_chunk` and if it gets back a `SnapshotItem`, it loads the contents into its `Blockstore` and finally sets the committed state.
+- `App::commit` calls `SnapshotClient::notify` with the current block height and the committed state parameters.
+- `tmconv::to_snapshot` maps the fields of a `SnapshotManifest` to the CometBFT type, in particular it encodes the `FvmStateParams` into the `metadata` field
+- `tmconv::from_snapshot` tries to restore a `SnapshotManifest` from an offer, comparing the CID of the decoded `FvmStateParams` (which is the `AppHash` of Fendermint) to the `app_hash` in the offer, thus checking the integrity that the offer is coming from a remote which, according to the Light Client, has a consensus that agrees with the snapshot (unless they lie about the snapshot contents).
```

Comment:
This style of docs is extremely useful to someone studying the code. It belongs in the module/crate-level Rustdocs rather than a separate markdown file.